### ISSUE TITLE
Reference images inside reference links

### ISSUE THREAD

**Issue by drwpow** [rsms/markdown-wasm#4]
Given the following markdown (this one found in [testing-library/react-testing-library](https://github.com/testing-library/react-testing-library)):

```md
[![Build Status][build-badge]][build]

[build-badge]: https://img.shields.io/travis/testing-library/react-testing-library.svg?style=flat-square
[build]: https://travis-ci.org/testing-library/react-testing-library
```

This is expected:

```html
<a href="https://travis-ci.org/testing-library/react-testing-library"><img src="https://img.shields.io/travis/testing-library/react-testing-library.svg?style=flat-square" alt="Build Status" /></a>
```

But instead the following markup is generated (link is after `<img>` tag, and extra square brackets are left):

```html
[<img src="https://img.shields.io/travis/testing-library/react-testing-library.svg?style=flat-square" alt="Build Status">]<a href="https://travis-ci.org/testing-library/react-testing-library">build</a>
```

While I can’t say I’ve ever used the pattern much, it seems that quite a number of packages on npm use this paradigm.

Would love to help on this issue if desired. Thanks!

**Comment by mity** [comment]
Fixed in MD4C upstream.

Next time, feel free to open issues there if it can be reproduced with it. :)

**Comment by rsms** [comment]
Fix patch from upstream (thank you @mity!) & published in [v1.1.2](https://github.com/rsms/markdown-wasm/releases/tag/v1.1.2)
